src/support_code_library_builder/validate_arguments.ts

```diff
@@ -1,30 +1,37 @@
 import _, { Dictionary } from 'lodash'
 import { doesNotHaveValue } from '../value_checker'
+import { DefineStepPattern, IDefineStepOptions } from './types'
 
 interface IValidation {
   identifier: string
   expectedType: string
   predicate: (args: any) => boolean
 }
 
+interface IDefineStepArguments {
+  pattern?: DefineStepPattern
+  options?: IDefineStepOptions
+  code?: Function
+}
+
 const optionsValidation = {
   expectedType: 'object or function',
-  predicate({ options }) {
+  predicate({ options }: Partial<IDefineStepArguments>) {
```

Comment:
I don't think we need `Partial` given all the fields are optional

Reply:
Yep, fixed